data.py

```diff
@@ -35,119 +36,185 @@ def set_context(resource_item):
-        g += Graph().parse(
-            data=json.dumps(set_context(datasets_source_dict)),
-            format="json-ld",
-        )
-    assert len(g) > graph_length
-    graph_length = len(g)
+
+    # Validate data and add to graph
+    validate_and_parse_json(g, schemas.Datasets, datasets_source_dict, "datasets")
+
+    # Extract edition URLs from datasets.json for validation later
+    dataset_editions_urls = [
+        dataset["editions_url"] for dataset in datasets_source_dict["datasets"]
+    ]
+
+    # Extract publishers from datasets.json for validation later
+    dataset_publishers = {
+        dataset["publisher"] for dataset in datasets_source_dict["datasets"]
+    }
 
     # ------------------
     # Editions resources
     # ------------------
 
-    # Load from disk
-    editions_source_path = Path(subbed_metadata_store_content_path / "editions")
-    # Validate then add to graph
-    schema = schemas.Editions
-    process_json_files(g, editions_source_path, schema)
+    # Load json files from disk
+    editions = process_json_files(editions_source_path)
+
+    # Empty dict to hold version data from editions for validation later
+    versions_in_edition = dict()
+    # Empty set to hold topic data from editions for validation later
+    topics_in_editions = set()
+
+    # Validate data and add to graph
+    for edition in editions:
+        assert (
+            edition["@id"] in dataset_editions_urls
+        ), f"Editions URL {edition['@id']} not found in {dataset_editions_urls}"
+        for edn in edition["editions"]:
+            summarised_editions = [
+                {
+                    "@id": edn["@id"],
+                    "issued": edn["issued"],
+                    "modified": edn["modified"],
+                }
+            ]
+            versions_in_edition[edn["versions_url"]] = edn["versions"]
+            assert (
+                edn["publisher"] in dataset_publishers
+            ), f"{edn['publisher']} not in publisher list {dataset_publishers}"
+            topics_in_editions.update(edn["topics"])
+        for dataset in datasets_source_dict["datasets"]:
+            if dataset["editions_url"] == edition["@id"]:
+                for dataset_edition in dataset["editions"]:
+                    assert (
+                        dataset_edition in summarised_editions
+                    ), f"Discrepancy between {dataset_edition} and {summarised_editions}"
+        validate_and_parse_json(g, schemas.Editions, edition, "editions")
 
     # ------------------
     # Versions resources
     # ------------------
 
-    versions_source_path = Path(
-        subbed_metadata_store_content_path / "editions/versions"
-    )
+    # Load json files from disk
+    versions = process_json_files(versions_source_path)
 
-    # Validate then add to graph
-    schema = schemas.Versions
-    process_json_files(g, versions_source_path, schema)
+    # Validate data and add to graph
+    for version in versions:
```

Comment:
Possibly more potential for splitting this function up here in the versions, topics and publishers resources.